inspirehep/modules/disambiguation/tasks.py

```diff
@@ -0,0 +1,191 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Celery tasks provided by Disambiguation module."""
+
+from __future__ import (
+    absolute_import,
+    division,
+    print_function,
+)
+
+from celery import current_app, shared_task
+from celery.utils.log import get_task_logger
+from sqlalchemy.orm.exc import StaleDataError
+
+from invenio_db import db
+from invenio_indexer.api import RecordIndexer
+from invenio_indexer.signals import before_record_index
+from invenio_records import Record
+
+from inspirehep.modules.disambiguation.beard import make_beard_clusters
+from inspirehep.modules.disambiguation.logic import process_clusters
+from inspirehep.modules.disambiguation.models import DisambiguationRecord
+from inspirehep.modules.disambiguation.search import (
+    create_record,
+    create_signatures,
+    get_blocks_from_record,
+    get_records_from_block,
+)
+from inspirehep.modules.disambiguation.receivers import (
+    append_updated_record_to_queue,
+)
+
+logger = get_task_logger(__name__)
+
+
+@shared_task(ignore_result=True)
+def disambiguation_daemon():
+    """Run disambiguation daemon as Celery task.
+
+    The method is invoked by Celery beat schedule and can be
+    configured from 'config.py'.
+
+    The method collects the records UUIDs from the database,
+    dispatch each of the phonetic blocks found within
+    these records to Celery workers and finally dumps the records.
+    """
+    phonetic_blocks = set()
+
+    # Get the records classified for clustering.
+    records_to_cluster = DisambiguationRecord.query.all()
+
+    # For each record get the phonetic blocks.
+    for record in records_to_cluster:
```

Comment:
Here you might have more times the same record. So you are going to call `get_blocks_from_record()` in the loop too many times. It's better if you can actually extract directly the list of record IDs involved and transform them into a set.